### PR TITLE
Change formatter from `autopep8` to `black` (string normalization separate commit).

### PR DESCRIPTION
**Pull request by hvy**
Basically same changes as in https://github.com/optuna/optuna/pull/1020 (ongoing, replacing `autopep8` with `black` and applying `black` to our existing codebase) but separated out string normalization changes to its own commit. Reviewing diffs should become slightly easier.

## Resolving/Avoiding conflicts derived from this PR

If possible, apply `black` to your changes before merging with the `master` branch as described below. Otherwise, apply `black` after resolving conflicts.

### How to apply `black`

See https://github.com/psf/black. In short, install `black` and run the following command to modify Python files in-place to conform to Black's style.

```bash
black <path to Python file/directory> --line-length 99
```

Append `--check` to not change the files in-place, but to instead return a status code whether the code is conformed or not.

You can also use `black` from within your editor. The procedure is described here https://black.readthedocs.io/en/stable/editor_integration.html#.

### CODE REVIEW

File: optuna/study.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def study_id(self):
         """
 
         message = (
-            'The use of `Study.study_id` is deprecated. ' 'Please use `Study.study_name` instead.'
+            "The use of `Study.study_id` is deprecated. " "Please use `Study.study_name` instead."
```
Comment (toshihikoyanase) [inline optuna/study.py]: I think these strings can be merged.

```suggestion
            "The use of `Study.study_id` is deprecated. Please use `Study.study_name` instead."
```

File: optuna/trial.py [hunk 2]
```diff
@@ -728,12 +728,12 @@ def trial_id(self):
         """
 
         warnings.warn(
-            'The use of `Trial.trial_id` is deprecated. ' 'Please use `Trial.number` instead.',
+            "The use of `Trial.trial_id` is deprecated. " "Please use `Trial.number` instead.",
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
            "The use of `Trial.trial_id` is deprecated. Please use `Trial.number` instead.",
```

File: optuna/trial.py [hunk 3]
```diff
@@ -805,7 +805,7 @@ def study_id(self):
             The study ID.
         """
 
-        message = 'The use of `Trial.study_id` is deprecated. ' 'Please use `Trial.study` instead.'
+        message = "The use of `Trial.study_id` is deprecated. " "Please use `Trial.study` instead."
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
        message = "The use of `Trial.study_id` is deprecated. Please use `Trial.study` instead."
```

File: optuna/trial.py [hunk 4]
```diff
@@ -728,12 +728,12 @@ def trial_id(self):
...
 
         warnings.warn(
-            'The use of `Trial.trial_id` is deprecated. ' 'Please use `Trial.number` instead.',
+            "The use of `Trial.trial_id` is deprecated. " "Please use `Trial.number` instead.",
             DeprecationWarning,
         )
 
         self.logger.warning(
-            'The use of `Trial.trial_id` is deprecated. ' 'Please use `Trial.number` instead.'
+            "The use of `Trial.trial_id` is deprecated. " "Please use `Trial.number` instead."
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
            "The use of `Trial.trial_id` is deprecated. Please use `Trial.number` instead."
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -57,10 +57,10 @@ jobs:
             pip install --progress-bar off .[checking]
 
       - run:
-          name: autopep8
+          name: black
           command: |
             . venv/bin/activate
-            autopep8 . -r --diff --exit-code
+            black . --line-length 99 --check --exclude="docs"
```
Comment (keisuke-umezawa) [inline .circleci/config.yml]: Fine. 
FYI
If we want to use a configration file, we can use `pyproject.toml` for black.

File: setup.cfg [hunk 6]
```diff
@@ -1,9 +1,6 @@
-# This section is for autopep8.
-[pep8]
-max-line-length = 99
-
 # This section is for flake8.
 [flake8]
+ignore = E203,E231,W503
```
Comment (keisuke-umezawa) [inline setup.cfg]: These ignore are due to black.

**Comment by keisuke-umezawa** [comment]
@hvy @toshihikoyanase @sile 
Thank you for this heavy task 🙏

File: .circleci/config.yml (hunk 5 above)
Comment (hvy) [reply .circleci/config.yml]: A drop by, leaving a reference https://github.com/optuna/optuna/pull/1020#discussion_r392775740

**Pull request merged**